File: src/main/java/me/narutopig/bors/listeners/RecipeUnlocker.java
package me.narutopig.bors.listeners;

import me.narutopig.bors.CustomRecipes;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.inventory.Recipe;

public class RecipeUnlocker implements Listener {
    @EventHandler
    public void onPlayerJoin(PlayerJoinEvent event) {
        Player player = event.getPlayer();

        System.out.println("potato");

        for (Recipe recipe : CustomRecipes.recipes) {
            player.discoverRecipe(recipe.getResult().getType().getKey());
        }
    }
}

File: src/main/kotlin/io/github/narutopig/listeners/RecipeUnlocker.kt
package io.github.narutopig.listeners

import io.github.narutopig.CustomRecipes
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

class RecipeUnlocker : Listener {
    @EventHandler
    fun onPlayerJoin(event: PlayerJoinEvent) {
        val player = event.player
        for (recipe in CustomRecipes.recipes) {
            player.discoverRecipe(recipe!!.result.type.key)
        }
    }
}